test/inputs-dynamodb-data-test.js

```diff
@@ -0,0 +1,305 @@
+require('./support');
+
+const dynamo = require('../lib/dynamo');
+const kinesis = require('../lib/kinesis');
+const logger = require('../lib/logger');
+const DynamodbData = require('../lib/inputs/dynamodb-data');
+
+describe('dynamodb-data', () => {
+  it('organizes records into payloads and segments', () => {
+    const recs = [
+      { listenerEpisode: 'le1', digest: 'd1', timestamp: 1000, type: 'antebytes', any: 'data' },
+      { listenerEpisode: 'le1', digest: 'd1', timestamp: 2000, type: 'bytes' },
+      { listenerEpisode: 'le2', digest: 'd1', timestamp: 3000, type: 'bytes' },
+      { listenerEpisode: 'le1', digest: 'd1', timestamp: 4000, type: 'segmentbytes', segment: 2 },
+    ];
+    const ddb = new DynamodbData(recs);
+
+    expect(ddb.payloads).to.eql({
+      'le1.d1': { timestamp: 1000, type: 'antebytes', any: 'data' },
+    });
+    expect(ddb.segments).to.eql({
+      'le1.d1': ['2000', '4000.2'],
+      'le2.d1': ['3000'],
+    });
+  });
+
+  it('recognizes antebytes and bytes records', () => {
+    const ddb = new DynamodbData();
+
+    expect(ddb.check({})).to.be.false;
+    expect(ddb.check({ type: 'combined' })).to.be.false;
+    expect(ddb.check({ type: 'postbytes' })).to.be.false;
+
+    expect(ddb.check({ type: 'antebytes' })).to.be.true;
+    expect(ddb.check({ type: 'antebytespreview' })).to.be.true;
+    expect(ddb.check({ type: 'bytes' })).to.be.true;
+    expect(ddb.check({ type: 'segmentbytes' })).to.be.true;
+  });
+
+  it('encodes payload keys', () => {
+    const ddb = new DynamodbData();
+    const data = { listenerEpisode: 'a', digest: 'b' };
+
+    expect(ddb.encodeKey(data)).to.eql(['a.b', {}]);
+    expect(ddb.encodeKey({ any: [9, 'data'], ...data })).to.eql(['a.b', { any: [9, 'data'] }]);
```

Comment:
Just wondering, what are these ddb rows called? If they had a name. Some intermediately recorded impression or download?

Reply:
Hm... until now I've just referred to them as the "payload".  But I've also called them the "redirect data" at various places in Dovetail / DovetailRouter.  And they also have `{ "type": "antebytes" }` (the record that comes _before_ the bytes), so sometimes you'll see that.

I guess I'm saying I'm inconsistent?  But things are getting better, now that we're almost done with non-IAB2 mode Dovetail (TAL was using it).  And it's been years since we used the "IAB2 Preview" mode for anything.